dhcpv4/nclient4/conn_unix.go

```diff
@@ -105,15 +105,18 @@ func (upc *BroadcastRawUDPConn) ReadFrom(b []byte) (int, net.Addr, error) {
 		}
 
 		ipHdr := ipv4(buf.Data())
+		headerLength := ipHdr.headerLength()
 
-		if !buf.Has(int(ipHdr.headerLength())) {
+		if !buf.Has(int(headerLength)) {
 			continue
 		}
 
-		ipHdr = ipv4(buf.Consume(int(ipHdr.headerLength())))
+		ipHdr = ipv4(buf.Consume(int(headerLength)))
 
-		if ipHdr.transportProtocol() != udpProtocolNumber {
-			continue
+		if headerLength > protocol {
+			if ipHdr.transportProtocol() != udpProtocolNumber {
+				continue
+			}
 		}
 
 		if !buf.Has(udpHdrLen) {
```

Comment:
How about solving the issue by simply moving this if before the `udpProtocolNumber` check?

Reply:
As I understand, `udpHdrLen` equals 8, so this won't solve the issue.